src/main/java/net/dries007/tfc/client/render/entity/SimpleMobRenderer.java

```diff
@@ -27,17 +30,24 @@
     private final ResourceLocation texture;
     @Nullable
     private final ResourceLocation babyTexture;
+    @Nullable
+    private final ResourceLocation maleTexture;
     private final Function<T, ResourceLocation> textureGetter;
     private final boolean doesFlop;
     private final float scale;
 
     public SimpleMobRenderer(EntityRendererProvider.Context ctx, M model, String name, float shadow, boolean flop, float scale, boolean hasBabyTexture, boolean itemInMouth, @Nullable Function<T, ResourceLocation> textureGetter)
+    {
+        this(ctx, model, name, shadow, flop, scale, hasBabyTexture, false, itemInMouth, textureGetter);
+    }
+    public SimpleMobRenderer(EntityRendererProvider.Context ctx, M model, String name, float shadow, boolean flop, float scale, boolean hasBabyTexture, boolean hasMaleTexture, boolean itemInMouth, @Nullable Function<T, ResourceLocation> textureGetter)
     {
         super(ctx, model, shadow);
         doesFlop = flop;
         texture = Helpers.animalTexture(name);
         babyTexture = hasBabyTexture ? Helpers.animalTexture(name + "_young") : null;
-        this.textureGetter = textureGetter != null ? textureGetter : e -> babyTexture != null && e.isBaby() ? babyTexture : texture;
+        maleTexture = hasMaleTexture ? Helpers.animalTexture(name + "_male") : null;
+        this.textureGetter = textureGetter != null ? textureGetter : e -> babyTexture != null && e.isBaby() ? babyTexture : maleTexture != null && (e instanceof WildAnimal) && ((WildAnimal) e).isMale() ? maleTexture : texture;
```

Comment:
this is getting too complicated. you can just use the texture getter. also check GenderedRenderAnimal and not WidlAnimal if that's what you're doing.